qiskit/extensions/rv.py

```diff
@@ -0,0 +1,75 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Rotation around an axis in x-y plane."""
+
+import numpy
+from qiskit.circuit.gate import Gate
+
+
+class RVGate(Gate):
+    r"""Rotation around arbitrary rotation axis v where |v| is
+    angle of rotation in radians.
+
+    **Circuit symbol:**
+
+    .. parsed-literal::
+
+             ┌──────┐
+        q_0: ┤ R(v) ├
+             └──────┘
+
+    **Matrix Representation:**
+
+    .. math::
+
+    \newcommand{\th}{|\vec{v}|}
+    \newcommand{\sinc}{\text{sinc}}
+        R(\vec{v}) = e^{-i \vec{v}\cdot\vec{\sigma}} =
+            \begin{pmatrix}
+                \cos{\th} -i v_z \sinc(\th) & -(i v_x + v_y) \sinc(\th) \\
+                -(i v_x - v_y) \sinc(\th) & \cos(\th) + i v_z \sinc(\th)
+            \end{pmatrix}
+    """
+
+    def __init__(self, v_x, v_y, v_z, basis='U3'):
```

Comment:
The last argument suggests that the basis can be chosen, but in `_define` it looks like U3 is required.

Reply:
fixed